File: ruby/space-age/space_age.rb
module BookKeeping
  VERSION = 1
end

class SpaceAge
  SECONDS = 31557600
  MERCURY = 0.2408467
  VENUS = 0.61519726
  MARS = 1.8808158
  JUPITER = 11.862615
  SATURN = 29.447498
  URANUS = 84.016846
  NEPTUNE = 164.79132

  def initialize(seconds)
    @earth_age = seconds.to_f / SECONDS
  end

  def on_earth
    @earth_age
  end

  def on_mercury
    @earth_age / MERCURY
  end

  def on_venus
    @earth_age / VENUS
  end

  def on_mars
    @earth_age / MARS
  end

  def on_jupiter
    @earth_age / JUPITER
  end

  def on_saturn
    @earth_age / SATURN
  end

  def on_uranus
    @earth_age / URANUS
  end

  def on_neptune
    @earth_age / NEPTUNE
  end
end

File: javascript/space-age/space-age.js
let SECONDS = 31557600;
let MERCURY = 0.2408467;
let VENUS = 0.61519726;
let MARS = 1.8808158;
let JUPITER = 11.862615;
let SATURN = 29.447498;
let URANUS = 84.016846;
let NEPTUNE = 164.79132;


let SpaceAge = function(seconds) {
  this.seconds = seconds;
  this.age = this.seconds / SECONDS;
}

SpaceAge.prototype.onEarth = function() {
  return Number((this.age).toFixed(2));
}
SpaceAge.prototype.onMercury = function() {
  return Number((this.age / MERCURY).toFixed(2));
}
SpaceAge.prototype.onVenus = function() {
  return Number((this.age / VENUS).toFixed(2));
}
SpaceAge.prototype.onMars = function() {
  return Number((this.age / MARS).toFixed(2));
}
SpaceAge.prototype.onJupiter = function() {
  return Number((this.age / JUPITER).toFixed(2));
}
SpaceAge.prototype.onSaturn = function() {
  return Number((this.age / SATURN).toFixed(2));
}
SpaceAge.prototype.onUranus = function() {
  return Number((this.age / URANUS).toFixed(2));
}
SpaceAge.prototype.onNeptune = function() {
  return Number((this.age / NEPTUNE).toFixed(2));
}

module.exports = SpaceAge;